### PR TITLE
Update schema descriptions

### PR DESCRIPTION
**Pull request by choldgraf**
This updates a few of the schema descriptions so that they are more clear.

- The `docker image` field now provides guidance for the structure that the value should take (to avoid confusion like in https://github.com/2i2c-org/infrastructure/issues/1032#issuecomment-1058674360).
- The descriptions of each user interface now re-use the content that is at try.jupyter.org .

### CODE REVIEW

File: frontend/src/schema.json [hunk 1]
```diff
@@ -6,7 +6,7 @@
       "type": "string",
       "title": "User docker image",
       "description": "Determines languages, libraries and interfaces available",
-      "help": "Leave this blank to use the default"
+      "help": "Should be of the form <docker registry>/<org>/<name>:<tag>. E.g., quay.io/2i2c/myimagename:cb5757e7c9f8. Leave this blank to use the default."
```
Comment (choldgraf) [inline frontend/src/schema.json]:
```suggestion
      "help": "For example: quay.io/2i2c/myimagename:cb5757e7c9f8. Leave this blank to use the default."
```

**Review by GeorgianaElena — APPROVED** [review]
🚀

**Review by damianavila — APPROVED** [review]

**Comment by yuvipanda** [comment]
This also needs the frontend js to be rebuilt I think before it can be deployed... Need to automate that.

**Review by damianavila — CHANGES_REQUESTED** [review]
As per @yuvipanda's last comment, this one needs the JS to be rebuilt.

From @GeorgianaElena 

> I generated this running npm run dev

**Comment by choldgraf** [comment]
OK I _think_ I just did this, let me know if I got it wrong.

While I was at it, I updated the contributing docs a bit, because they were a little unclear to me :-)

**Comment by manics** [comment]
> Need to automate that.

I had a go at copying https://github.com/jupyterhub/nbgitpuller/pull/211 + https://github.com/jupyterhub/nbgitpuller/pull/222 to this repo (untested, and has some bugs):
https://github.com/yuvipanda/jupyterhub-configurator/compare/main...manics:package?expand=1

It's a bit more complicated than nbgitpuller because of the extra webpack/babel/css config and subdirectory.

I was also puzzled by index.html:
https://github.com/yuvipanda/jupyterhub-configurator/blob/996405d2a7017153d5abe592b8028fed7a1801bb/jupyterhub_configurator/static/index.html#L1-L15

**Comment by choldgraf** [comment]
@manics I found it confusing too haha, that's why I tried to document a little bit more.

This feels like something that `nox` or `tox` could be useful for. Then a person could just run something like `nox -s compile` and it would (1) install `npm` in a virtual environment if needed, (2) run the proper `npm` commands to build the assets needed. We use this in the Jupyter Book world a lot and it helps reduce confusion associated with installing NPM environments

**Comment by manics** [comment]
The idea behind Jupyter-packaging (or the equivalent code) is to include the npm build commands in setup.py so the usual `pip install .` takes care of almost everything, and you never need to commit the generated JS assets- they're generated and packaged automatically into the wheel.

**Comment by yuvipanda** [comment]
@manics wow that would be *awesome*. Can you make a PR too?

**Comment by manics** [comment]
Will do when I get time to test it (assuming it works!)

**Comment by choldgraf** [comment]
@manics you should check out the Sphinx Theme Builder that @pradyunsg built, it uses the `pyproject.toml` build standard to include an npm environment with an install. It is lovely: https://github.com/pradyunsg/sphinx-theme-builder

**Comment by manics** [comment]
`pyproject.toml`/pep517  is nice, but if you're using setuptools you'll loose the option of an editable install (`pip install -e .`) :cry: https://github.com/pypa/setuptools/issues/2816 

I've therefore gone for a build without pep517 for now in https://github.com/yuvipanda/jupyterhub-configurator/pull/13
If you merge your current PR I can update the docs in my PR

**Comment by choldgraf** [comment]
@manics yeah, we have switched to using flit for builds in jupyter book, which gives you editable builds 🙂

For this PR, I think it's good to go? I don't know that I can merge into @yuvipanda 's repo though

**Comment by pradyunsg** [comment]
> `pyproject.toml`/pep517 is nice, but if you're using setuptools you'll loose the option of an editable install (`pip install -e .`) 😢 [pypa/setuptools#2816](https://github.com/pypa/setuptools/issues/2816)

Yes, but if you're doing something like this in a custom build backend, you won't be relying on setuptools to perform the build for you. Basically, instead of:

> The idea behind Jupyter-packaging (or the equivalent code) is to include the npm build commands in setup.py so the usual `pip install .`

You'd end up doing for sphinx-theme-builder is doing -- running npm commands _before_ packaging things up yourself.

**Comment by choldgraf** [comment]
Can somebody tell me if there's anything else I should do, or if we can merge this?

**Comment by yuvipanda** [comment]
Unfortunately this project is an architectural dead end - see https://github.com/2i2c-org/features/issues/26. Going to close this one out.

**Pull request closed without merging**